cmd/monitor/monitor.go

```diff
@@ -129,7 +153,16 @@ func prependLatestBlocks(ctx context.Context, ms *monitorStatus, rpc *ethrpc.Cli
 	}
 }
 
-func appendOlderBlocks(ctx context.Context, ms *monitorStatus, rpc *ethrpc.Client) {
+func appendOlderBlocks(ctx context.Context, ms *monitorStatus, rpc *ethrpc.Client) error {
+	if ms.MinBlockRetrieved == nil {
+		log.Warn().Msg("Nil min block")
+		return fmt.Errorf("the min block is nil")
+	}
+	if !currentlyFetchingHistoryLock.TryLock() {
+		return fmt.Errorf("the function is currently locked")
+	}
+	defer currentlyFetchingHistoryLock.Unlock()
```

Comment:
lock 👍

Reply:
so previously there were multiple fetches going on..?

Reply:
Yeah I believe so
`fetchBlocks` is called in a loop forever and by default it was always calling `appendOlderBlocks`. That was it's own issue. But it seemed possible that `fetchBlocks` -> `appendOlderBlocks` could be called in parallel with `renderMonitorUI` -> `redraw` -> `appendOlderBlocks`

I was also considering running `appendOlderBlocks` in a different go routined in order to avoid blocking the UI while we wait for blocks to load. If we do that then it's possible that `appendOlderBlocks` could be called multiple times while pressing the down arrow. 

I didn't end up implementing that, but if we do it in the future it should make `appendOlderBlocks` work safely (I hope)